pkg/logging/log.go

```diff
@@ -43,13 +44,17 @@ func Init(flags *flag.FlagSet) {
 
 // Setup configures the logger with the supplied log format.
 // It returns an error if the JSON logger could not be initialized or passed logFormat is not recognized.
-func Setup(logFormat string) error {
+func Setup(logFormat string, level int) error {
 	switch logFormat {
 	case TextFormat:
 		// in text mode we use FormatSerialize format
 		globalLog = klogr.New()
 	case JSONFormat:
-		zapLog, err := zap.NewProduction()
+		zc := zap.NewProductionConfig()
+		// Zap's levels get more and less verbose as the number gets smaller and higher respectively (DebugLevel is -1, InfoLevel is 0, WarnLevel is 1, and so on).
+		zc.Level = zap.NewAtomicLevelAt(zapcore.Level(-1 * level))
+		zc.EncoderConfig.NameKey = "name"
```

Comment:
What is `NameKey` ?

Reply:
@eddycharly this is to add the proper name field as we talked about:

without setting that in zap, the logger name is called ""logger
```
{"level":"info","ts":1667933714.795587,"logger":"config","caller":"config/config.go:11","msg":"hello world"}
```
but when the Namekey is set, it's now called name as we expect
```
{"level":"info","ts":1667933732.179918,"name":"config","caller":"config/config.go:11","msg":"hello"}
```

Reply:
Why `name` is better than `logger` in this case ?
What happens if there is a `value` with the `name` key ?

Reply:
> What happens if there is a value with the name key ?

Do you know the answer to this question ?

Reply:
there will be duplicate keys

Reply:
or probably the former one get's overwritten?

Reply:
We can't have duplicate keys in JSON.

Reply:
or the key gets overwritten

Reply:
yeah

Reply:
```go
l := logging.WithName("configger").WithValues("name", "secname")
l.Info("hello world")
```

{"level":"info","ts":1668001906.6426702,"name":"configger","caller":"config/config.go:11","msg":"hello world","name":"secname"}

Reply:
outch

Reply:
oh yeah, it doesn't doesn't allow duplicates..i was testing with the wrong function call

Reply:
What do you mean ?

Reply:
@eddycharly 
```go
l := logging.WithName("configger").WithValues("name", "secname")
l.Info("hello world")
```
{"level":"info","ts":1668002280.555058,"name":"configger","caller":"config/config.go:11","msg":"hello world","name":"secname"}

Reply:
name is duplicate?

Reply:
Any chance that values can get into a separate field ?
Something like:
`"values": { "key1": "value1", "key2": "value2" }`

Reply:
@eddycharly can we resolve this then..as i've removed the name key

Reply:
looks good
```json
{"level":"info","ts":1668002690.357266,"logger":"configger","caller":"config/config.go:12","msg":"hello world","name":"secname"}
```

Reply:
If we can put key/values in a dedicated field it would be even better.

Reply:
@yimikao are you working on this ? Or shall we merge what we have here ?

Reply:
@eddycharly how i can do this in a simpler way

i want to do this but i cant marshal map[interface{}]interface{}
```go
func WithValues(keysAndValues ...interface{}) logr.Logger {
	m := make(map[interface{}]interface{})
	for i := 0; i < len(keysAndValues); i += 2 {
		m[keysAndValues[i]] = keysAndValues[i+1]
	}
	byt, _ := json.Marshal(m)
	return GlobalLogger().WithValues("values", string(byt))
}
```

Reply:
We can't do it this way, it has to happen somewhere in zap or zapr.
Let's merge what we have here and see if we can improve later.

Reply:
@eddycharly i think we can merge this for now

Reply:
It's in the merge queue...